### PR TITLE
Document breaking changes in dotnet test for .NET 7

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Document the breaking changes in the options available for `dotnet test` both in the list of breaking changes of .NET 7 and in the documentation page of the `dotnet test`.

Relates to:

- https://github.com/dotnet/sdk/issues/21952
- https://github.com/dotnet/sdk/issues/21389
- https://github.com/dotnet/sdk/issues/28986

### CODE REVIEW

File: docs/core/tools/dotnet-test.md [hunk 1]
```diff
@@ -80,10 +80,18 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
 
 ## Options
 
-- **`-a|--test-adapter-path <ADAPTER_PATH>`**
+> [!WARNING]
+> Breaking changes in options:
```
Comment (MarcoRossignoli) [inline docs/core/tools/dotnet-test.md:84]: should we add "starting from?"

Comment (Evangelink) [reply docs/core/tools/dotnet-test.md:84]: Yes! I was thinking about it and missed it.

**Comment by Evangelink** [comment]
@gewarren For the `dotnet test` page, my idea is to try to improve discoverability and not rely only on breaking changes web page. If you are thinking of a better or more uniform to some other doc page way of doing it, please feel free to modify my PR.

**Review by MarcoRossignoli — APPROVED** [review]

**Review by tdykstra — APPROVED** [review]
Looks good, thanks.

File: docs/core/compatibility/7.0.md [hunk 2]
```diff
@@ -110,6 +110,8 @@ If you're migrating an app to .NET 7, the breaking changes listed here might aff
 | - | :-: | :-: | - |
 | [Version requirements for .NET 7 SDK](sdk/7.0/vs-msbuild-version.md) | ✔️ | ✔️ | 7.0.100 |
 | [MSBuild serialization of custom types in .NET 7](sdk/7.0/custom-serialization.md) | ❌ | ❌ | 7.0.100 |
+| [dotnet test: switch `-a` to alias `--arch` instead of `--test-adapter-path`](https://github.com/dotnet/sdk/issues/21389) | ❌ | ❌ | Preview 1 |
```
Comment (gewarren) [inline docs/core/compatibility/7.0.md:113]: @PriyaPurkayastha Are you okay with these entries on the overview page linking off to GitHub issues?

**Pull request merged**